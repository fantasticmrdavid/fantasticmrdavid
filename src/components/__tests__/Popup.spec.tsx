import React from 'react';
import { render } from "@testing-library/react";
import Ui from 'components/Popup/ui';
import {
  Close,
  Landscape,
  Lightbox,
  Portrait,
} from 'components/Popup/styles';

const props = {
  isActive: true,
  onClose: jest.fn(),
  loading: false,
  orientation: 'landscape',
  title: 'Popup Test',
  url: '/about',
};

describe('Component -> Popup', () => {
  it('should render the closed Popup UI without throwing an error', () => {
    expect(render(<Ui {...props} isActive={false} />)).toMatchSnapshot();
  });

  it('should render the open Popup UI without throwing an error', () => {
    expect(render(<Ui {...props} />)).toMatchSnapshot();
  });

  it('should call close() when Lightbox is clicked', () => {
    const component = render(<Ui {...props} />);
    component.getByTestId("popupLightbox").click();
    expect(props.onClose).toHaveBeenCalled();
  });

  it('should call close() when Close element is clicked', () => {
    const component = render(<Ui {...props} />);
    component.getByTestId("popupClose").click();
    expect(props.onClose).toHaveBeenCalled();
  });

  it('should render a Landscape element if a URL is provided and orientation is landscape', () => {
    const component = render(<Ui {...props} />);
    expect(component.getByTestId("popupImage")).toHaveAttribute("data-orientation", "landscape");
  });

  it('should NOT render a Portrait element if the orientation is landscape', () => {
    const component = render(<Ui {...props} />);
    expect(component.getByTestId("popupImage")).not.toHaveAttribute("data-orientation", "portrait");
  });

  it('should render a Portrait element if a URL is provided and orientation is portrait', () => {
    const component = render(<Ui {...props} orientation="portrait" />);
    expect(component.getByTestId("popupImage")).toHaveAttribute("data-orientation", "portrait");
  });

  it('should NOT render a Landscape element if the orientation is landscape', () => {
    const component = render(<Ui {...props} orientation="portrait" />);
    expect(component.getByTestId("popupImage")).not.toHaveAttribute("data-orientation", "landscape");
  });
});
