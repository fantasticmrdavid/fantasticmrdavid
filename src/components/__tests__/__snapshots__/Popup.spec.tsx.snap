// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component -> Popup should render the closed Popup UI without throwing an error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="styles__Container-sc-hsnn8o-0 itzZCe"
      >
        <div
          class="styles__Lightbox-sc-hsnn8o-1 iCWwzy"
          data-testid="popupLightbox"
        />
        <div
          class="styles__Dialog-sc-hsnn8o-2 kkIep"
        >
          <div
            class="styles__Close-sc-hsnn8o-5 jEoFoT"
            data-testid="popupClose"
          >
            X
          </div>
          <img
            alt="Popup Test"
            class="styles__Landscape-sc-hsnn8o-3 dDwluL"
            data-orientation="landscape"
            data-testid="popupImage"
            src="/about"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="styles__Container-sc-hsnn8o-0 itzZCe"
    >
      <div
        class="styles__Lightbox-sc-hsnn8o-1 iCWwzy"
        data-testid="popupLightbox"
      />
      <div
        class="styles__Dialog-sc-hsnn8o-2 kkIep"
      >
        <div
          class="styles__Close-sc-hsnn8o-5 jEoFoT"
          data-testid="popupClose"
        >
          X
        </div>
        <img
          alt="Popup Test"
          class="styles__Landscape-sc-hsnn8o-3 dDwluL"
          data-orientation="landscape"
          data-testid="popupImage"
          src="/about"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Component -> Popup should render the open Popup UI without throwing an error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="styles__Container-sc-hsnn8o-0 jkHdDZ"
      >
        <div
          class="styles__Lightbox-sc-hsnn8o-1 iCWwzy"
          data-testid="popupLightbox"
        />
        <div
          class="styles__Dialog-sc-hsnn8o-2 kkIep"
        >
          <div
            class="styles__Close-sc-hsnn8o-5 jEoFoT"
            data-testid="popupClose"
          >
            X
          </div>
          <img
            alt="Popup Test"
            class="styles__Landscape-sc-hsnn8o-3 dDwluL"
            data-orientation="landscape"
            data-testid="popupImage"
            src="/about"
          />
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="styles__Container-sc-hsnn8o-0 jkHdDZ"
    >
      <div
        class="styles__Lightbox-sc-hsnn8o-1 iCWwzy"
        data-testid="popupLightbox"
      />
      <div
        class="styles__Dialog-sc-hsnn8o-2 kkIep"
      >
        <div
          class="styles__Close-sc-hsnn8o-5 jEoFoT"
          data-testid="popupClose"
        >
          X
        </div>
        <img
          alt="Popup Test"
          class="styles__Landscape-sc-hsnn8o-3 dDwluL"
          data-orientation="landscape"
          data-testid="popupImage"
          src="/about"
        />
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
